### ISSUE TITLE
[4.0] After creating an article, modified date is set but no modified by user

### ISSUE THREAD

**Issue by Bakual** [joomla/joomla-cms#16368]
### Steps to reproduce the issue
* Create a new article, contact or newsfeed (maybe others as well) and save once intially
* Check the "Modified Date" and "Modified By"

### Expected result
* Either both are not set like in J3 where they are only set after the next save (edit)
* or both are set

### Actual result
Only the modified date is set, but the user remains empty until the next save (edit)

### Additional comments
I can see where that comes from, since with MySQL strict mode the modified column can't be an empty string. It has to be a valid date or the "nulldate".
However apparently it somehow works for com_fields where creating a field has the same behavior as in J3: The modified date is only populated after the first successive save/edit. I think it's because com_fields, the modified date has `filter="unset"` in the XML which means it gets nulled during validation.

We should choose a way how we want to handle that date and be consistent across all extensions, either always setting it (but then also the user) or only after an edit (like com_fields).

**Comment by tonypartridge** [comment]
This PR should solve the issue:

https://github.com/joomla/joomla-cms/pull/11530

As it should be merged into 4.0 too.

**Comment by Bakual** [comment]
I don't think it does fix the underlying issue, because I think it would still not accept an empty string or null.

**Comment by Bakual** [comment]
Closing as we have a PR #16444
